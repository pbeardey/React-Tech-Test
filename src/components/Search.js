// src/components/Search.js
import React, { useState } from "react";
import getImages from "../requests/getImages";
import "../styles/search.css";

const Search = ({ setSearchResults }) => {
  const [searchText, setSearchText] = useState("")
 
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  }

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(searchText));
  }

  return (
    <div className="search">
    <form className="search-form" onSubmit={handleSubmit}>
      <input className="search-input" type="text" onChange={handleInputChange}></input>  
      <button className="search-button" type="submit">Go!</button>
    </form>
    </div>
  )
}

export default Search;