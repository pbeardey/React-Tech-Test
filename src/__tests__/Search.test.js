// src/__tests__/Search.test.js

import { render, screen } from '@testing-library/react';
import Search from '../components/Search';

describe("Search", () => {
  
  it("renders Search", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders textbox for Search", () => {
    render(<Search />);
    const inputBox = screen.getByRole('textbox');
    expect(inputBox).toBeInTheDocument();
  });
});