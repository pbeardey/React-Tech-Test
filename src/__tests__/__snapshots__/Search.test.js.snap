// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders Search 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
